packages/agent/src/connect.ts

```diff
@@ -0,0 +1,179 @@
+import { DwnProtocolDefinition, DwnRecordsPermissionScope } from './index.js';
+import {
+  Web5ConnectAuthResponse,
+  Oidc,
+  type PushedAuthResponse,
+} from './oidc.js';
+import { pollWithTtl } from './utils.js';
+import { DidDht } from '@web5/dids';
+
+async function initClient({
+  connectServerUrl,
+  walletUri,
+  permissionRequests,
+  onWalletUriReady,
+  validatePin,
+}: WalletConnectOptions) {
+  // ephemeral client did for ECDH, signing, verification
+  // TODO: use separate keys for ECDH vs. sign/verify. could maybe use secp256k1.
+  const clientDid = await DidDht.create();
```

Comment:
Sanity: Does this need to be a `did:dht` or can we use a `did:jwk`?

Reply:
It would be better if the DIDs returned to the client are DID JWKs

Reply:
fixed